core/spec.md

```diff
@@ -2031,45 +2070,31 @@ new attributes within the model definitions themselves), but only if
-compatibility of clients or to ensure existing entities do not need to be
-changed to be consistent with the new model.
-
-Implementations are REQUIRED to ensure that after any model changes are made,
-all of the entities in the Registry are valid with respect to the new model
-definition (including all Versions of Resources). How this is achieved will
-be implementation specific. For example, implementations can choose to
-automatically modify existing entities, or even to delete non-conforming
-entities (such as when Groups or Resource types are removed). However, it is
-STRONGLY RECOMMENDED that implementations not delete entities due to attribute
-modifications.
-
-When model changes are made, it might be necessary to modify existing entity's
-attributes to ensure they are conformant with the new model. The following
-ordered suggestions are defined to provide consistency across implementations,
-but it is NOT REQUIRED for implementations to follow these rules if a
-different set of changes are more appropriate:
-- If a `default` value is defined for the attribute, then it SHOULD be used.
-- If the attribute is OPTIONAL, then the attribute SHOULD be deleted.
-- If an enum is defined and `strict` is `true`, the first `enum` value SHOULD
-  be used.
-- If valid, the zero value for the attribute's type SHOULD be used:
-  - Array, Map, Object: empty value (e.g. `[]`, `{}`, `{}` respectively).
-  - Boolean: false.
-  - Numeric: zero.
-  - String, URI, URIReference, URI-Template, URL: empty string (`""`).
-  - Timestamp: zero (00:00:00 UTC Jan 1, 1970).
-- If valid, String attributes SHOULD use `"undefined"`.
-
-Changes to the model MUST NOT include changing the `hasDocument` aspect of a
-Resource from `false` to `true` if there are attributes defined on that
-Resource whose names conflict with the `RESOURCE*` attributes. Any attempt do
-to so MUST generate an error.
-
-If a backwards incompatible change is needed, and the existing entities need
-to be preserved, then it is RECOMMENDED that users define a new Group or
-Resource types for future instances of those entities.
+Once a Registry has been created, changes to the model are limited to:
+- Adding new Groups or Resources.
+- Adding new OPTIONAL attributes.
+
+Any model change that might result in existing data within the Registry
+becoming non-compliant MUST generate an error, even if it does not do so. For
+example, the following non-exhaustive list of changes MUST generate an error:
+- Deleting an attribute, Group or Resource.
+- Changing the `name` or `type` of an attribute.
+- Removing a value from the enumeration list of an attribute.
+- Enabling/disabling a feature that might impact storing data
+  (e.g. `hasdocument`).
+
+Some example changes that are allowed:
+- Changing the `description` of an attribute.
+- Adding an enumeration value to an attribute.
+- Changing the `readonly` aspect or `default` value of an attribute.
```

Comment:
Wouldn't changing the `readonly` aspect from false to true possibly break something?

Reply:
hmm would it? I agree it would change what future API calls might do to the data, but would any existing data become invalid?